chapter_recurrent-neural-networks/bptt.md

```diff
@@ -140,21 +154,21 @@ $$\begin{aligned}
+A number of things follow from this potentially very intimidating expression. Firstly, it pays to store intermediate results, i.e., powers of $\mathbf{W}_{hh}$ as we work our way through the terms of the loss function $L$. Secondly, this simple linear example already exhibits some key problems of long sequence models: it involves potentially very large powers $\mathbf{W}_{hh}^j$. In it, eigenvalues which are smaller than $1$ vanish for large $j$ and eigenvalues which are larger than $1$ diverge. This is numerically unstable and gives undue importance to potentially irrelvant past detail. One way to address this is to truncate the sum at a computationally convenient size. Later on in this chapter we will see how more sophisticated sequence models such as LSTMs can alleviate this further. In practice, this truncation is effected by *detaching* the gradient after a given number of steps.
 
 ## Summary
 
-* Back-propagation through time is merely an application of backprop to sequence models with a hidden state.
+* Backpropagation through time is merely an application of backprop to sequence models with a hidden state.
 * Truncation is needed for computational convenience and numerical stability.
-* High powers of matrices can lead top divergent and vanishing eigenvalues. This manifests itself in the form of exploding or vanishing gradients.
-* For efficient computation intermediate values are cached.
+* High powers of matrices can lead to divergent and diminishing eigenvalues, which further results in gradients exploding or vanishing.
+* For efficient computation, intermediate values are cached.
 
 ## Exercises
 
-1. Assume that we have a symmetric matrix $\mathbf{M} \in \mathbb{R}^{n \times n}$ with eigenvalues $\lambda_i$. Without loss of generality assume that they are ordered in ascending order $\lambda_i \leq \lambda_{i+1}$. Show that $\mathbf{M}^k$ has eigenvalues $\lambda_i^k$.
-1. Prove that for a random vector $\mathbf{x} \in \mathbb{R}^n$ with high probability $\mathbf{M}^k \mathbf{x}$ will by very much aligned with the largest eigenvector $\mathbf{v}_n$ of $\mathbf{M}$. Formalize this statement.
+1. Assume that we have a symmetric matrix $\mathbf{M} \in \mathbb{R}^{n \times n}$ with eigenvalues $\lambda_i$. Without loss of generality, assume that they are ordered in an ascending order $\lambda_i \leq \lambda_{i+1}$. Show that $\mathbf{M}^k$ has eigenvalues $\lambda_i^k$.
+1. Prove that for a random vector $\mathbf{x} \in \mathbb{R}^n$ with high probability $\mathbf{M}^k$, $\mathbf{x}$ will be very much likely aligned with the largest eigenvector $\mathbf{v}_n$ of $\mathbf{M}$. Formalize this statement.
```

Comment:
for XXX, with high probability YYY will be .........

Reply:
My fault